tests/solver.rs

```diff
@@ -1429,6 +1432,203 @@ fn test_explicit_root_requirements() {
+
+    let a_version_set = provider.intern_version_set(&a_spec);
+    let b_version_set = provider.intern_version_set(&b_spec);
+    let c_version_set = provider.intern_version_set(&c_spec);
+    let d_version_set = provider.intern_version_set(&d_spec);
+
+    let cond_req1 = ConditionalRequirement::new(a_version_set, b_version_set.into());
+    let cond_req2 = ConditionalRequirement::new(b_version_set, c_version_set.into());
+    let cond_req3 = ConditionalRequirement::new(c_version_set, d_version_set.into());
+
+    let requirements = vec![
+        cond_req1,
+        cond_req2,
+        cond_req3,
+        a_version_set.into(), // Require package a
+    ];
+
+    let mut solver = Solver::new(provider);
+    let problem = Problem::new().requirements(requirements);
+    let solved = solver.solve(problem).unwrap();
+    let result = transaction_to_string(solver.provider(), &solved);
+    // All packages should be installed due to chain reaction
+    insta::assert_snapshot!(result, @r###"
+        a=1
+        b=1
+        c=1
+        d=1
+        "###);
+}
+
+#[test]
+fn test_multiple_conditions_same_package() {
+    let mut provider = BundleBoxProvider::new();
+
+    // Setup packages
+    provider.add_package("a", 1.into(), &[], &[]);
+    provider.add_package("b", 1.into(), &[], &[]);
+    provider.add_package("c", 1.into(), &[], &[]);
+    provider.add_package("target", 1.into(), &[], &[]);
+
+    // Create multiple conditions that all require the same package
+    let a_spec = Spec::parse_union("a 1").next().unwrap().unwrap();
+    let b_spec = Spec::parse_union("b 1").next().unwrap().unwrap();
+    let c_spec = Spec::parse_union("c 1").next().unwrap().unwrap();
+    let target_spec = Spec::parse_union("target 1").next().unwrap().unwrap();
+
+    let a_version_set = provider.intern_version_set(&a_spec);
+    let b_version_set = provider.intern_version_set(&b_spec);
+    let c_version_set = provider.intern_version_set(&c_spec);
+    let target_version_set = provider.intern_version_set(&target_spec);
+
+    // If any of a, b, or c is installed, require target
+    let cond_req1 = ConditionalRequirement::new(a_version_set, target_version_set.into());
+    let cond_req2 = ConditionalRequirement::new(b_version_set, target_version_set.into());
+    let cond_req3 = ConditionalRequirement::new(c_version_set, target_version_set.into());
+
+    let requirements = vec![
+        cond_req1,
+        cond_req2,
+        cond_req3,
+        b_version_set.into(), // Only require package b
+    ];
+
+    let mut solver = Solver::new(provider);
+    let problem = Problem::new().requirements(requirements);
+    let solved = solver.solve(problem).unwrap();
+    let result = transaction_to_string(solver.provider(), &solved);
+    // b and target should be installed
+    insta::assert_snapshot!(result, @r###"
+        b=1
+        target=1
+        "###);
+}
+
+#[test]
+fn test_circular_conditional_dependencies() {
```

Comment:
I think you should add a few tests that include more versions for a single package.